openfe/utils/visualization.py

```diff
@@ -115,20 +118,26 @@ def _draw_molecules(
     highlight_color: Tuple[float, float, float, float]
         RGBA tuple for the default highlight color used in the mapping
         visualization
+    atom_mapping: Dict[Tuple[int,int], Dict[int, int]], optional
+        used to align the molecules to each othter for clearer visualization,
+        default None
     """
     if d2d is None:
         # select default layout based on number of molecules
-        grid_x, grid_y = {
-            1: (1, 1),
-            2: (2, 1),
-        }[len(mols)]
-        d2d = Chem.Draw.rdMolDraw2D.MolDraw2DCairo(grid_x*300, grid_y*300,
-                                                   300, 300)
+        grid_x, grid_y = {1: (1, 1), 2: (2, 1), }[len(mols)]
+        d2d = Chem.Draw.rdMolDraw2D.MolDraw2DCairo(
+            grid_x * 300, grid_y * 300, 300, 300)
 
     # squash to 2D
-    copies = [copy.deepcopy(mol) for mol in mols]
-    for mol in copies:
-        AllChem.Compute2DCoords(mol)
+    copies = [Chem.Mol(mol) for mol in mols]
+    [AllChem.Compute2DCoords(mol) for mol in copies]
+
+    if atom_mapping is not None and len(atom_mapping) > 0:
```

Comment:
`len` check is meaningless here; looping over `items` for an empty dict will just do nothing.

Generally, I think the better approach is to normalize `atom_mapping` to always be a dict by the time you get here with an earlier statement like:

```python
if atom_mapping is None:
    atom_mapping = {}
```

I'd usually put that at right at the start of the function. Then you've normalized/standardized your inputs, and don't have to worry about special cases later. This reduces the cognitive load on a reader looking at later parts of your function.

The result would be that the entire `if` statement here, which handles special cases, instead gets handled by making so that by the time you get here, special cases aren't special. No need for the `if`.

(There's general philosophy that anything that involves deeper nesting increases cognitive load readers of your code, so avoiding nested methods is good. This idea comes from the concept of [cognitive complexity](https://docs.codeclimate.com/docs/cognitive-complexity), which I think of as an easier-to-estimate cousin to the more formally interesting [cyclomatic complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity).)